src/models/account.ts

```diff
@@ -1,13 +1,30 @@
 import RestfulModel from './restful-model';
 import Attributes, { Attribute } from './attributes';
 import NylasConnection from '../nylas-connection';
+import { NativeAuthenticationProvider } from './connect';
+
+export enum AuthStatusState {
+  Valid = 'valid',
+  Invalid = 'invalid',
+}
+
+export type AuthStatusProperties = {
+  state: AuthStatusState;
+  provider: NativeAuthenticationProvider;
+  scopes: string;
+  error_message?: string;
+};
 
 export type AccountProperties = {
   name: string;
   emailAddress: string;
   provider: string;
   organizationUnit: string;
+  /**
+   * @deprecated will be removed in future versions. use `authStatus` instead
```

Comment:
This is actually not finalized. SyncState will likely not be deprecated but mapped to AuthStatus, and drop things that don't make a lot of sense like `partial` and a few others. AuthStatus would be an additional attribute. The reason is this a major breaking change would require so many downstream changes for our customers that the ROI to introduce a breaking change is not worth it.